projectile.el

```diff
@@ -3300,6 +3300,13 @@ entirely."
   :risky t
   :package-version '(projectile "0.12.0"))
 
+(defun projectile-find-file-hook-function ()
+  "Called by `find-file-hook' when `projectile-mode' is on."
+  (unless (file-remote-p default-directory)
```

Comment:
I got a nice speedup with just this change. Maybe fixing the lighter would speed it up further, but this change is needed anyway, I think.

Of course, some people might prefer for the hook to work anyway on the remote, if their remote is fast. So a custom var can be added for this.

But adding 1 function to `find-file-hook` is in any case better than adding 3. Much simpler to understand and customize.